### PR TITLE
Improve re-registration

### LINKED ISSUE #471: Improve re-registration behavior on already installed systems
This is a follow up of #434 

In that issue we enabled re-registration at each boot, however:
- We only try once and in case of failure we abort
- We do not handle the "multiple machines re-booting all at the same time" scenarios, so in some context with hundreds of devices it won't scale well
- We are making a distinction between "live system" and "installed system", which adds complexity. Ideally we should retry consistently without having to think about the installation state

One possible solution is to actually remove logic from the `elemental-register` code and rely on systemd instead, for example by defining [StartLimitIntervalSec and StartLimitBurst](https://www.freedesktop.org/software/systemd/man/systemd.unit.html#StartLimitIntervalSec=interval).  

This however won't really cover the "submarine coming to surface after months" scenario.  
For that we may need to actually implement an exponential backoff so that we can keep trying (at max once every 30 min?) indefinitely.

### PR DESCRIPTION
**Pull request by anmazzotti**
Resolve #471  

- Added full registration config and statefile path parameters on elemental-register
- Remove support for multiple configuration files
- Added (hardcoded) timer to skip registration updates for 24 hours
- Store emulated TPM seed for future registrations
- Exit with error code in case of failures (systemd will manage restarts)
- Use virtual filesystem where possible

### CODE REVIEW

File: pkg/install/install.go
Comment (anmazzotti) [inline pkg/install/install.go:1]: This is all moved code, nothing new here.

File: cmd/register/main.go
Comment (anmazzotti) [inline cmd/register/main.go:1]: I strongly suggest reviewing the new file ignoring the diff.

**Review by davidcassany — APPROVED** [review]
Nice, well done, looks good to me :+1:

File: cmd/register/main.go [hunk 1]
```diff
@@ -17,416 +17,171 @@ limitations under the License.
...
+	state, err := stateHandler.Load()
 	if err != nil {
-		return "", err
+		return false, fmt.Errorf("loading registration state")
 	}
-
-	log.Debugf("Decoded CloudConfig:\n%s\n", string(bytes))
-	_, err = f.Write(bytes)
-	return f.Name(), err
+	return !state.HasLastUpdateElapsed(registrationUpdateSuppressTimer), nil
```
Comment (davidcassany) [inline cmd/register/main.go:169]: I just realized this only relies on the 24h delay time frame. Shouldn't we consider reboots as a valid case to register regardless of the timing of the last register? as context could have changed on a reboot.

Comment (anmazzotti) [reply cmd/register/main.go:169]: Opened a new issue: https://github.com/rancher/elemental-operator/issues/481

**Pull request merged**